src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala

```diff
@@ -19,9 +23,256 @@ class AppDeployIntegrationTest
   //clean up state before running the test case
   before(cleanUp())
 
-  def extractDeploymentIds(app: RestResult[AppDefinition]): Seq[String] = {
-    for (deployment <- (app.entityJson \ "deployments").as[JsArray].value)
-      yield (deployment \ "id").as[String]
+  test("create a simple app without health checks") {
+    Given("a new app")
+    val app = appProxy(testBasePath / "app", "v1", instances = 1, withHealth = false)
+
+    When("The app is deployed")
+    val result = marathon.createApp(app)
+
+    Then("The app is created")
+    result.code should be (201) //Created
+    extractDeploymentIds(result) should have size 1
+    waitForEvent("deployment_success")
+  }
+
+  test("create a simple app with http health checks") {
+    Given("a new app")
+    val app = appProxy(testBasePath / "http-app", "v1", instances = 1, withHealth = false).
+      copy(healthChecks = Set(HealthCheck(gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)))
+
+    When("The app is deployed")
+    val result = marathon.createApp(app)
+
+    Then("The app is created")
+    result.code should be (201) //Created
+    extractDeploymentIds(result) should have size 1
+    waitForEvent("deployment_success")
+  }
+
+  test("create a simple app with tcp health checks") {
+    Given("a new app")
+    val app = appProxy(testBasePath / "tcp-app", "v1", instances = 1, withHealth = false).
+      copy(healthChecks = Set(HealthCheck(protocol = Protocol.TCP, gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)))
+
+    When("The app is deployed")
+    val result = marathon.createApp(app)
+
+    Then("The app is created")
+    result.code should be (201) //Created
+    extractDeploymentIds(result) should have size 1
+    waitForEvent("deployment_success")
+  }
+
+  test("create a simple app with command health checks") {
+    Given("a new app")
+    val app = appProxy(testBasePath / "command-app", "v1", instances = 1, withHealth = false).
+      copy(healthChecks = Set(HealthCheck(protocol = Protocol.COMMAND, command = Some(Command("true")), gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)))
```

Comment:
Quite a long line. It's actually slightly annoying reviewing it like this.

But if you disagree heavily, I do not care that much.